api/v1/routes/payments.py

```diff
@@ -0,0 +1,80 @@
+from fastapi import Depends, APIRouter, status, Query
+from sqlalchemy.orm import Session
+from typing import Annotated
+
+from api.utils.success_response import success_response
+from api.v1.schemas.payment import PaymentListResponse
+from api.v1.services.payment import payment_service
+from api.v1.services.user import user_service
+from api.db.database import get_db
+from api.v1.models import User
+
+payment = APIRouter(prefix='/payments', tags=['Payments'])
+
+@payment.get('/current-user', status_code=status.HTTP_200_OK, response_model=PaymentListResponse)
+def get_payments_for_current_user(
+    current_user: Annotated[User, Depends(user_service.get_current_user)],
+    limit: Annotated[int, Query(ge=1, description="Number of payments per page")] = 10,
+    page: Annotated[int, Query(ge=1, description="Page number (starts from 1)")] = 1,
+    db: Session = Depends(get_db), 
+):
+    '''
+    Endpoint to retrieve a paginated list of payments of ``current_user``.
+
+    Query parameter: 
+        - limit: Number of payment per page (default: 10, minimum: 1)
+        - page: Page number (starts from 1)
+    '''
+    if not current_user:
```

Comment:
There is no need for this check. Remove it

Reply:
Removed @joboy-dev